package/lib/compileFunctions.js

```diff
@@ -47,10 +47,10 @@ module.exports = {
         'nodejs8';
       funcTemplate.properties.timeout =
         _.get(funcObject, 'timeout') || _.get(this, 'serverless.service.provider.timeout') || '60s';
-      funcTemplate.properties.environmentVariables = _.merge(
-        {},
-        _.get(this, 'serverless.service.provider.environment'),
-        funcObject.environment // eslint-disable-line comma-dangle
+      funcTemplate.properties.environmentVariables = _.transform(
```

Comment:
It'll be nicer to use `_.mapValues`

Reply:
yeah, I initially used that approach, but thought that it would be useful to have the key name in the error message as well so the user would know which key was configured incorrectly. I wasn't sure how to easily access keys from `mapValues` for the error message, so I went with transform.

If including the env var name feels unnecessary, I can update to use mapValues for sure

Reply:
In `mapValues` callback name is provided as second argument, and it's cleaner as you just return coerced value and do not need to overwrite it manually

Reply:
ah okay. sorry about my confusion there - I had misread the docs. Thanks for talking me through this. Just pushed an update.